src/pages/stale-pr.tsx

```diff
@@ -1,30 +1,117 @@
 import { FC, useState, useEffect } from 'react';
 import Head from '@/components/head';
-import Section from '@/components/pullRequests/section';
 import Layout from '@/components/Layout';
+import PullRequest from '@/components/pullRequests';
 import fetch from '@/helperFunctions/fetch';
+import CardShimmer from '@/components/Loaders/cardShimmer';
+
+type pullRequestType = {
+  title: string;
+  username: string;
+  createdAt: string;
+  updatedAt: string;
+  url: string;
+};
 
 const stalePR: FC = () => {
-  const [pullRequests, setPullRequests] = useState<[]>([]);
+  const [pullRequests, setPullRequests] = useState<pullRequestType[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [page, setPage] = useState(1);
+
+  async function fetchStalePRs() {
+    setLoading(true);
+    const response = await fetch({
+      url: `${process.env.NEXT_PUBLIC_BASE_URL}/pullrequests/stale?page=${page}`,
+    });
+    setPullRequests(response.data.pullRequests);
+    setLoading(false);
+  }
+
+  function handlePrev() {
+    setPage(page > 1 ? page - 1 : 1);
+  }
+
+  function handleNext() {
+    setPage(pullRequests.length ? page + 1 : page);
+  }
 
   useEffect(() => {
-    (async () => {
-      const res = await fetch({
-        url: `${process.env.NEXT_PUBLIC_BASE_URL}/pullrequests/stale`,
-      });
-      setPullRequests(res.data.pullRequests);
-    })();
-  }, []);
+    fetchStalePRs();
+  }, [page]);
+
+  const getPRs = () => pullRequests.map((pullRequest: pullRequestType) => {
+    const {
+      title, username, createdAt, updatedAt, url: link,
+    } = pullRequest;
+    return (
+      <>
+        <PullRequest
+          key={link}
+          title={title}
+          username={username}
+          createdAt={createdAt}
+          updatedAt={updatedAt}
+          url={link}
+        />
+      </>
+    );
+  });
 
   return (
-    <div>
-      <Layout>
-        <Head title="Stale PRs" />
-        <div className="container">
-          <Section heading="Stale PRs" content={pullRequests} />
-        </div>
-      </Layout>
-    </div>
+    <Layout>
+      <Head title="Stale PRs" />
+
+      <div className="container">
+        {loading
+          ? [...Array(10)].map((e: number) => <CardShimmer key={e} />)
+          : getPRs()}
+      </div>
+      {!pullRequests.length ? <p className="center-text">No more stale PRs...</p> : ''}
+
+      <div className="pagination">
+        <button
+          className="pagination-btn"
+          type="button"
+          onClick={handlePrev}
+          disabled={page <= 1}
+        >
+          Prev
+        </button>
+
+        <button
+          className="pagination-btn"
+          type="button"
+          onClick={handleNext}
+          disabled={pullRequests.length === 0}
+        >
+          Next
+        </button>
+      </div>
```

Comment:
The similar code we have on `pages/openPRs.tsx`
why not refactor this and add it in a component and call that component at both pages?